eval/compile.rkt

```diff
@@ -189,6 +189,10 @@
        [(and (even? (numerator y)) (odd? (denominator y))) `(pow (fabs ,x) ,y)]
        [(and (odd? (numerator y)) (odd? (denominator y))) `(copysign (pow (fabs ,x) ,y) ,x)]
        [else `(pow ,x ,y)])]
+
+    ; Some simplifications to prevent overflow
+    [`(log (exp ,x)) x]
+    [`(log (pow ,x ,y)) `(* ,y (log ,x))]
```

Comment:
Isn't this one not correct? For, like `(log (pow -2 2))`? Maybe if we rewrite to `(* y (log (fabs x)))` that might be correct? But gotta sit and think this through.

Reply:
Yeah, or what if `(log (pow -2 3))` it should not be `(* y (log (fabs x)))` it instead should be `(* y (log x))` or something since it needs to produce an `illegal-interval`. 
This negative case for `x` is kinda weird.

Reply:
Oh right yeah, for negatives we gotta produce an error. Ugh. Well, so yeah, that's that. I suppose we could compile to some ugly `if` construction, dunno if that will actually be any faster / better.

Reply:
I agree, probably it's better to just remove this one. We do not know at compile time whether `x` is negative or whether `y` is odd or even. Just will drop this one

Reply:
We can also add this rule: `(exp (log x)) -> (ensure x > 0)`, where `ensure` will return `x` if true, illegal interval if `x > 0 = (ival #f #f)` and `x` with a possible error flag if `x > 0 = (ival #t #f)`

Reply:
But assert returns a boolean interval, while original `x` may be needed

Reply:
Oh, right. There's a `then` operator to help with cases like this:

    (then (assert (> x 0)) x)

will first check if `x > 0` and, if so, return `x`.

Reply:
Ahh, this rewrite `(exp (log x)) -> (then (assert (> x 0)) x)` is kinda silly, it will never overflow. Anyways, probably it's good to have this simplification